src/bucket/Bucket.cpp

```diff
@@ -238,14 +250,14 @@ Bucket::loadPoolShareTrustLinessByAccount(
     // Get upper and lower bound for poolshare trustline range associated
     // with this account
     auto searchRange = getIndex().getPoolshareTrustlineRange(accountID);
-    if (searchRange.first == 0)
+    if (searchRange.first == 0 && searchRange.second == 0)
```

Comment:
Isn't this actually a protocol change?

Reply:
No, if there are no poolshare trustline keys in a bucket, `getPoolshareTrustlineRange` returns the pair {0, 0}. We only checked the first 0, so this hardens the check slightly. This pair is the file offsets within the bucket to scan for the trustline entries. In practice, it should be impossible for any `LedgerEntry` to start at an offset of 0, because the first entry should always be a `MetaEntry`. This is why checking just the first member of the pair is sufficient. The first possible offset for a `LedgerEntry` should always by 16. This just hardens the check a little, since `getPoolshareTrustlineRange` is expected to return {0, 0} if no keys exist.

Reply:
I don't think we can make assumptions regarding offset validity: earlier versions of buckets do not support meta entries, so offset 0 is possible. It would be nice to make the API less error-prone (and fix the offset 0 issue) by explicitly returning a nullopt if there are no trustline keys in the bucket